scaleway/resource_iot_device.go

```diff
@@ -242,12 +244,26 @@ func resourceScalewayIotDeviceCreate(ctx context.Context, d *schema.ResourceData
 
 	d.SetId(newRegionalIDString(region, res.Device.ID))
 
-	// Certificate and Key cannot be retreived later
-	cert := map[string]interface{}{
-		"crt": res.Certificate.Crt,
-		"key": res.Certificate.Key,
+	// If user certifcate is provided.
+	if devCrt, ok := d.GetOk("certificate.0.crt"); ok {
+		// Set user certificate to device.
+		// It cannot currently be added in the create device request.
+		_, err := iotAPI.SetDeviceCertificate(&iot.SetDeviceCertificateRequest{
+			Region:         region,
+			DeviceID:       res.Device.ID,
+			CertificatePem: devCrt.(string),
```

Comment:
Please add a safer method to cast this. e.g.
`s, ok := devCrt.(string)
if !ok {
    //  manage your error
}`

Reply:
As discussed, for coherency with the basecode we will let as is for now and take it in consideration for future improvements.